src/components/NewStack.js

```diff
@@ -126,24 +127,36 @@ class NewStack extends PureComponent {
 
   onSubmit (e) {
     e.preventDefault()
+    console.log(e)
 
     let updateOperationsState = false
 
-    const operations = this.state.operations.map((operation) => {
-      const required = this.props.operations[operation.name].required || []
-      const addedOptions = Object.keys(operation.options)
-      const missing = required.filter((field) => {
-        const index = addedOptions.indexOf(field)
-        return index === -1 || !operation.options[field].length
-      })
+    const ajv = new Ajv({
+      allErrors: true
+    })
 
-      operation.errors = {}
-      missing.forEach((field) => {
-        operation.errors[field] = `${field} is required`
-      })
+    this.operationValidators = {}
+    Object.keys(this.props.operations).forEach(key => {
+      const operation = this.props.operations[key]
+      if (key === 'noop' && Array.isArray(operation.properties)) {
+        operation.properties = {}
+      }
+      this.operationValidators[key] = ajv.compile(operation)
+    })
 
-      if (missing.length) {
+    const operations = this.state.operations.map(operation => {
+      this.operationValidators[operation.name](operation.options)
+      if (!this.operationValidators[operation.name](operation.options)) {
```

Comment:
you execute `this.operationValidators[operation.name](operation.options)` twice. You can just assign a variable to the return value and then check this. E.g.:
```
const valid = this.operationValidators[operation.name](operation.options)
if (!valid) {
  // .. code
}
```